File: pyshell/test_script/sleeper.py
#!/usr/bin/env python3
import signal
import os
import sys


def handler(signum):
    print("Handle signal {}".format(signal.getsignal(signum)))
    signal.signal(signum, signal.SIG_DFL)
    time.sleep(1)
    print("KILLING")
    os.kill(os.getpid(), signum)

if __name__ == "__main__":
    fd = open("/dev/pts/7", "w")
    print("Start : {}".format(sys.argv[0]), file=fd)
    signal.signal(signal.SIGTSTP, handler)
    signal.sigwait([signal.SIGTSTP])

File: pyshell/test_script/sleeper.c
#include <signal.h>
#include <unistd.h>
#include <stdio.h>

char *name;

void	handler(int signum)
{
    printf("%s: Handle signal %d\n", name, signum);
    signal(signum, SIG_DFL);
    sleep(1);
    printf("KILLING\n");
    if (kill(getpid(), signum) == -1)
		perror("kill");
    printf("out\n");
}

int		main(int argc, char **argv)
{
	name = argv[0];
	printf("Start : %s\n", argv[0]);
	signal(SIGTSTP, handler);
	pause();
	printf("end\n");

}
